File: chap01/GuitarStore-r1-enum/Guitar.cpp
#include "Guitar.h"

Guitar::Guitar(){}  
Guitar::Guitar(string serialNumber, double price, 
        GBuilder builder, string model,GType type,
        GWood backWood, GWood topWood){

    serialNumber_ = serialNumber;
    price_ = price;
    builder_ = builder.toString();
    model_ = model;
    type_ = type.toString();
    backWood_ = backWood.toString();
    topWood_ = topWood.toString();
}
const string Guitar::getSerialNumber() const{
    return serialNumber_;
}

const double Guitar::getPrice() const{
    return price_;
}
void Guitar::setPrice(float newPrice){
    price_ = newPrice;
}
const string Guitar::getBuilder() const{
    return builder_;
}
const string Guitar::getModel() const{
    return model_;
}
const string Guitar::getType() const{
    return type_;
}
const string Guitar::getBackWood() const{
    return builder_;
}
const string Guitar::getTopWood() const{
    return topWood_;
}

File: src/chapter1/Guitar.java
package chapter1;

public class Guitar {
    private String serialNumber;
    private String builder;
    private String model;
    private String backWood;
    private String topWood;
    private String type;

    public double getPrice() {
        return price;
    }

    private double price;

    public String getSerialNumber() {
        return this.serialNumber;
    }

    public String getBuilder() {
        return this.builder;
    }


    public String getModel() {
        return this.model;
    }

    public String getBackWood() {
        return this.backWood;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public String getTopWood() {
        return this.topWood;
    }

    public String getType() {
        return this.type;
    }


    public Guitar(String serialNumber, double price,
                  String builder, String model, String type,
                  String backWood, String topWood) {
        this.serialNumber = serialNumber;
        this.price = price;
        this.builder = builder;
        this.model = model;
        this.type = type;
        this.backWood = backWood;
        this.topWood = topWood;
    }

}